$MirOS: ports/multimedia/mplayer/patches/patch-libmpdemux_demuxer_c,v 1.2 2005/05/19 21:44:58 tg Exp $
$OpenBSD: patch-libmpdemux_demuxer_c,v 1.2 2005/09/04 23:51:52 espie Exp $
--- libmpdemux/demuxer.c.orig	Sun Apr  3 14:08:04 2005
+++ libmpdemux/demuxer.c	Wed Dec 28 19:38:41 2005
@@ -25,6 +25,10 @@
 // new_sh_video don't match aids and vids it accepts from the command line
 int demux_aid_vid_mismatch = 0;
 
+#ifdef STREAMING_LIVE_DOT_COM
+#include "demux_rtp.h"
+#endif
+
 void free_demuxer_stream(demux_stream_t *ds){
     ds_free_packs(ds);
     free(ds);
@@ -1350,7 +1354,12 @@ switch(file_format){
 
  case DEMUXER_TYPE_MPEG_TY: 
  case DEMUXER_TYPE_MPEG_PS: {
-  sh_video=d_video->sh;sh_video->ds=d_video;
+  if (d_video)
+    sh_video=d_video->sh;
+  else
+    sh_video = NULL;
+  if (sh_video)
+    sh_video->ds=d_video;
 //  if(demuxer->stream->type!=STREAMTYPE_VCD) demuxer->movi_start=0; // for VCD
 
   if(audio_id!=-2) {
@@ -1404,8 +1413,8 @@ pts_from_bps=0; // !!!
 if ((sh_video=demuxer->video->sh) && sh_video->bih)
   mp_msg(MSGT_DEMUX,MSGL_INFO,"VIDEO:  [%.4s]  %ldx%ld  %dbpp  %5.3f fps  %5.1f kbps (%4.1f kbyte/s)\n",
     (char *)&sh_video->bih->biCompression,
-    sh_video->bih->biWidth,
-    sh_video->bih->biHeight,
+    (long) sh_video->bih->biWidth,
+    (long) sh_video->bih->biHeight,
     sh_video->bih->biBitCount,
     sh_video->fps,
     sh_video->i_bps*0.008f,
